aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -63,6 +67,19 @@ def setup(self):
         self.ctx.is_converged = False
         self.ctx.iteration = 0
 
+        if self.inputs.final_scf and 'base_final_scf' in self.inputs:
+            raise ValueError('cannot specify `final_scf=True` and `base_final_scf` at the same time.')
+        elif self.inputs.final_scf:
+            import warnings
+            from aiida.common.warnings import AiidaDeprecationWarning
+            warnings.warn(
```

Comment:
Thanks for the review! To your question, no, for deprecation warnings we use the `warnings` module. We use the `AiidaDeprecationWarning` because normal deprecation warnings get swallowed. by using the subclass, we are sure it ends up in stdout so users are guaranteed to see it and can adapt their code.

Reply:
Hmm, but when and where does this warning show up? I'm actively looking for it and can't find it. 😅

Reply:
it should show up when you launch a `PwRelaxWorkChain` and use the `final_scf` input

Reply:
I don't see it, I'm afraid. Isn't the `setup` step in the `outline` only executed once the `PwRelaxWorkChain` is picked up by the daemon?

Reply:
Ah yeah of course, I forgot that it was in one of the outline steps and not the `define`. In that case, it will end up in the daemon log. Can check it with `verdi daemon logshow`. It is a good point actually, because people might not always look there. Maybe we can use a validator on the `final_scf` input port instead. That way the code is triggered upon launch and it will be visible when the user actually submits it.

Reply:
Good to know about the `verdi daemon logshow` command! I don't think we already have a section in the documentation on debugging? This would be a good command to explain in such a section. 🙂

A validator sounds like the right approach. Once you add it I'll give it a whirl! (and review the documentation on validators 😅 )

Reply:
Ok, I moved the deprecation warning to the validator. It should now show up on launch. Please have a look to verify